[netcdf-c] wrong HDF5 linkage

**Problem**
```
CMake Error at G:/MPINew/mpi_vcpkg/scripts/buildsystems/cmake/vcpkg-add_library.cmake:6 (_add_library):
  Target "netcdfhdf5" links to target "hdf5::hdf5-shared" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?
Call Stack (most recent call first):
  G:/MPINew/mpi_vcpkg/scripts/buildsystems/cmake/vcpkg-add_library.cmake:29 (vcpkg_add_library)
  libhdf5/CMakeLists.txt:19 (add_library)


CMake Error at G:/MPINew/mpi_vcpkg/scripts/buildsystems/cmake/vcpkg-add_library.cmake:6 (_add_library):
  Target "netcdfhdf5" links to target "hdf5::hdf5_hl-shared" but the target
  was not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?
Call Stack (most recent call first):
  G:/MPINew/mpi_vcpkg/scripts/buildsystems/cmake/vcpkg-add_library.cmake:29 (vcpkg_add_library)
  libhdf5/CMakeLists.txt:19 (add_library)

```
In a mixed static/shared build (hdf5 static/netcdf-c shared)

**Root of Problem**
https://github.com/microsoft/vcpkg/blob/6c7cb5dfdbe8932e589913d7c99943d6f004467c/ports/netcdf-c/hdf5_3.patch#L10-L12

**Probable solution**
```
if(HDF5_BUILD_SHARED_LIBS) 
 target_link_libraries(netcdfhdf5 PRIVATE hdf5::hdf5-shared hdf5::hdf5_hl-shared)
else()
 target_link_libraries(netcdfhdf5 PRIVATE hdf5::hdf5-static hdf5::hdf5_hl-static) 
endif() 
```

--- a/ports/netcdf-c/CONTROL
+++ b/ports/netcdf-c/CONTROL
@@ -1,5 +1,5 @@
 Source: netcdf-c
-Version: 4.7.0-3
+Version: 4.7.0-4
 Build-Depends: hdf5, curl
 Homepage: https://github.com/Unidata/netcdf-c
 Description: a set of self-describing, machine-independent data formats that support the creation, access, and sharing of array-oriented scientific data.

--- a/ports/netcdf-c/hdf5_3.patch
+++ b/ports/netcdf-c/hdf5_3.patch
@@ -2,11 +2,13 @@ diff --git a/libhdf5/CMakeLists.txt b/libhdf5/CMakeLists.txt
 index f3c7bbc..34fc2ab 100644
 --- a/libhdf5/CMakeLists.txt
 +++ b/libhdf5/CMakeLists.txt
-@@ -20,3 +20,7 @@ add_library(netcdfhdf5 OBJECT ${libnchdf5_SOURCES})
+@@ -20,3 +20,9 @@ add_library(netcdfhdf5 OBJECT ${libnchdf5_SOURCES})
  
  # Remember to package this file for CMake builds.
  ADD_EXTRA_DIST(${libnchdf5_SOURCES} CMakeLists.txt)
 +
-+if(BUILD_SHARED_LIBS)
-+target_link_libraries(netcdfhdf5 PRIVATE hdf5::hdf5-shared hdf5::hdf5-static hdf5::hdf5_hl-shared hdf5::hdf5_hl-static)
++if(HDF5_BUILD_SHARED_LIBS) 
++ target_link_libraries(netcdfhdf5 PRIVATE hdf5::hdf5-shared hdf5::hdf5_hl-shared)
++else()
++ target_link_libraries(netcdfhdf5 PRIVATE hdf5::hdf5-static hdf5::hdf5_hl-static) 
 +endif()

